### ISSUE TITLE
Gradient with opacity not working...

### ISSUE THREAD

**Issue by equinox2k** [SixLabors/ImageSharp#1040]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Gradient's with alpha not working as expected

### Steps to Reproduce

I have created an example that shows gradient not working as expected.. it should do a iOS style gradient overlay, however produces a semi transparent overlay but of a single color where alpha doesnt change...

```
using SixLabors.ImageSharp;
using SixLabors.ImageSharp.PixelFormats;
using SixLabors.ImageSharp.Processing;
using SixLabors.Primitives;
using SixLabors.Shapes;

namespace EQL.ImageProcessors
{
    public static class GlossProcessor
    {
        public static IImageProcessingContext ApplyGloss(this IImageProcessingContext ctx)
        {
            Size size = ctx.GetCurrentSize();
            IPathCollection glossPath = BuildGloss(size.Width, size.Height);
            var graphicsOptions = new GraphicsOptions(true)
            {
                ColorBlendingMode = PixelColorBlendingMode.Normal,
                AlphaCompositionMode = PixelAlphaCompositionMode.SrcAtop
            };
            var linearGradientBrush = new LinearGradientBrush(new Point(0, 0), new Point(0, size.Height / 2), GradientRepetitionMode.Repeat, new ColorStop(0, Color.White.WithAlpha(0.5f)), new ColorStop(1, Color.White.WithAlpha(0.25f)));
            return ctx.Fill(graphicsOptions, linearGradientBrush, glossPath);
        }

        private static IPathCollection BuildGloss(int imageWidth, int imageHeight)
        {
            var pathBuilder = new PathBuilder();
            pathBuilder.AddLine(new PointF(0, 0), new PointF(imageWidth, 0));
            pathBuilder.AddLine(new PointF(imageWidth, 0), new PointF(imageWidth, imageHeight * 0.4f));
            pathBuilder.AddBezier(new PointF(imageWidth, imageHeight * 0.4f), new PointF(imageWidth / 2, imageHeight * 0.6f), new PointF(0, imageHeight * 0.4f));
            pathBuilder.CloseFigure();
            return new PathCollection(pathBuilder.Build());
        }
    }
}
```

can be called by loading a image and applying via... 

image.Mutate(i => i.ApplyGloss());

### System Configuration
macOS 

- ImageSharp version:
0.7beta

**Comment by equinox2k** [comment]
@jongleur1983

**Comment by jongleur1983** [comment]
@equinox2k not exactly sure how if I can reproduce it correctly. Could you provide reference output and actual (wrong) output for some sample please? e.g. just a opaque red image and how it should look like would be great.

**Comment by equinox2k** [comment]
@jongleur1983, Here is the expected result...

![expected](https://user-images.githubusercontent.com/655939/67957226-870c6e00-fbb2-11e9-8110-8c70caf30189.png)

And actual...

![actual](https://user-images.githubusercontent.com/655939/67957227-870c6e00-fbb2-11e9-8e60-59737047b7f8.png)

**Comment by equinox2k** [comment]
Code used from above and actual image created using...
```
            using (var image = new Image<Rgba32>(1024, 1024))
            {
                image.Mutate(i => i.Fill(Color.Red));
                image.Mutate(i => i.ApplyGloss());
                image.Save("actual.png");
            }
```

**Comment by JimBobSquarePants** [comment]
Fixed in #1046
